smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ShapeValueGenerator.kt

```diff
@@ -229,22 +229,16 @@ class ShapeValueGenerator(
         override fun arrayNode(node: ArrayNode) {
             when (currShape.type) {
                 ShapeType.DOCUMENT -> {
-                    writer
-                        .writeInline("#T(\n", RuntimeTypes.Core.Smithy.Document)
-                        .indent()
-                        .writeInline("listOf(\n")
-                        .indent()
-
-                    node.elements.forEach {
-                        generator.writeShapeValueInline(writer, currShape, it)
-                        writer.writeInline(",\n")
+                    writer.withInlineBlock("#T(", ")", RuntimeTypes.Core.Smithy.Document) {
+                        writer.withInlineBlock("listOf(", ")") {
+                            node.elements.forEachIndexed { index, it ->
+                                generator.writeShapeValueInline(writer, currShape, it)
+                                if (index < node.elements.size - 1) {
+                                    writer.writeInline(",\n")
+                                }
```

Comment:
**Style**: There's no reason to omit the final comma. You could simplify this to the original `forEach` loop that always adds the trailer.